croquemort/migrations.py

```diff
@@ -28,3 +29,59 @@ def split_content_types(self):
             content_type = data.get('content-type')
             if content_type and ';' in content_type:
                 self.storage.store_content_type(url_hash, content_type)
+
+    @rpc
+    def add_hash_prefixes(self):
+        """
+        Add url and group hash prefixes where they are missing:
+        - /urls[<uhash>] -> /urls[<u:uhash>]
+        - /<uhash> -> /<u:uhash>
+        - /<u:uhash>/group=<ghash> -> /<u:uhash>/group=<g:ghash>
+        - /<frequency>[<ghash>] -> /<frequency>[<g:ghash>]
+        - /<ghash> -> /<g:ghash>
+        - /<g:ghash>/<uhash> -> /<g:ghash>/<u:uhash>
+
+        NB1: checks are not migrated because they expire quite fast
+        NB2: should be idempotent
+        """
+        log('Adding hash prefixes')
+        database = self.storage.database
+        url_prefix = '{}:'.format(HASH_PREFIXES['url'])
+        group_prefix = '{}:'.format(HASH_PREFIXES['group'])
+        idx = 0
+        # migrate url hashes in hash keys and urls list and migrate group key
+        # in url hash
+        for url_hash, data in self.storage.get_all_urls():
+            if not url_hash.startswith(url_prefix):
+                new_hash = '{}{}'.format(url_prefix, url_hash)
+                database.lset('urls', idx, new_hash)
+                database.rename(url_hash, new_hash)
+                if 'group' in data:
+                    new_g_hash = '{}{}'.format(group_prefix, data['group'])
+                    database.hset(new_hash, 'group', new_g_hash)
+            idx += 1
+        # migrate group hashes and nested url hashes
+        _, all_keys = database.scan()
+        for key in all_keys:
+            if database.type(key) == 'hash' \
+                    and not key.startswith('cache') \
```

Comment:
It's for a cache decorator used in views. I should probably migrate it too, come to think of it.